crates/loader/src/local/mod.rs

```diff
@@ -72,7 +72,11 @@ async fn prepare(
     let info = info(raw.info, &src);
 
     let mut config_root = raw.config.unwrap_or_default();
+    let component_ids: Vec<String> = raw.components.iter().map(|c| c.id.clone()).collect();
```

Comment:
instead of the `contains` call below, there's probably a way to just make sure these ids are unique, and error if they're not